web/modules/api/src/api.ts

```diff
@@ -71,34 +71,60 @@ class WikijumpAPIInstance extends Api<void> {
       const res = await login(data, requestParams)
       this._CSRF = res.csrf
       authSet(true)
+      this.updateIdentity()
       return res
     }
 
     this.authLogout = async requestParams => {
       await logout(requestParams)
       authSet(false)
+      this.updateIdentity()
     }
 
     this.authRefresh = async requestParams => {
       const res = await refresh(requestParams)
       this._CSRF = res.csrf
+      this.updateIdentity()
       return res
     }
 
     this.authCheck = async requestParams => {
       const res = await check(requestParams)
       authSet(res.authed)
+      this.updateIdentity()
       return res
     }
 
     this.accountRegister = async (data, requestParams) => {
       const res = await register(data, requestParams)
       this._CSRF = res.csrf
       authSet(true)
+      this.updateIdentity()
       return res
     }
   }
 
+  /**
+   * Updates the current client `UserIdentity`. Usually called when
+   * authentication state changes.
+   */
+  private async updateIdentity() {
```

Comment:
Not sure I fully understand what this does.

Reply:
It's just getting the client's current `UserIdentity` object (the one in the API schema) and setting a store's value to the retrieved identity. It looks kind of complicated, but it's just to make sure the clientside API can't get itself stuck in a weird state.